src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -410,7 +410,11 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
 
                     var wd = startupInfo.WorkingDirectory;
                     if (!IsRemote && !string.IsNullOrEmpty(wd) && wd.HasReadPermissions()) {
-                        await evaluation.SetWorkingDirectoryAsync(wd);
+                        try {
+                            await evaluation.SetWorkingDirectoryAsync(wd);
+                        } catch(REvaluationException) {
+                            await evaluation.SetDefaultWorkingDirectoryAsync();
```

Comment:
The only thing that can come from` EvaluateAsync` (which is what it ultimately does) is `REvaluationException` and `OperationCanceledException` - if there's something else, it's a bug in and of itself.